src/calendar/DatePicker.ts

```diff
@@ -0,0 +1,368 @@
+	private _openMonthPopup() {
+		const { onPopupChange } = this.properties;
+		this._monthPopupOpen = true;
+		this._callMonthPopupFocus = true;
+		this._yearPopupOpen = false;
+		this.invalidate();
+		onPopupChange && onPopupChange(this._getPopupState());
+	}
+
+	private _openYearPopup() {
+		const { onPopupChange } = this.properties;
+		this._yearPopupOpen = true;
+		this._callYearPopupFocus = true;
+		this._monthPopupOpen = false;
+		this.invalidate();
+		onPopupChange && onPopupChange(this._getPopupState());
+	}
+
+	private _renderMonthRadios() {
+		const { month } = this.properties;
+
+		return this.properties.monthNames.map((monthName, i) => v('label', {
+			key: `${this._idBase}_month_radios_${i}`,
+			classes: this.classes(css.monthRadio, i === month ? css.monthRadioChecked : null)
+		}, [
+			v('input', {
+				checked: i === month,
+				classes: this.classes(css.monthRadioInput),
+				name: `${this._idBase}_month_radios`,
+				tabIndex: this._monthPopupOpen ? 0 : -1,
+				type: 'radio',
+				value: i + '',
+				onchange: this._onMonthRadioChange,
+				onmouseup: this._closeMonthPopup
+			}),
+			v('abbr', {
+				classes: this.classes(css.monthRadioLabel),
+				title: monthName.long
+			}, [ monthName.short ])
+		]));
+	}
+
+	private _renderYearRadios() {
+		const { year } = this.properties;
+		const radios = [];
+
+		const yearLimits = this._getYearRange();
+		for (let i = yearLimits.first; i < yearLimits.last; i++) {
+			radios.push(v('label', {
+				key: `${this._idBase}_year_radios_${i}`,
+				classes: this.classes(css.yearRadio, i === year ? css.yearRadioChecked : null)
+			}, [
+				v('input', {
+					checked: i === year,
+					classes: this.classes(css.yearRadioInput),
+					name: `${this._idBase}_year_radios`,
+					tabIndex: this._yearPopupOpen ? 0 : -1,
+					type: 'radio',
+					value: i + '',
+					onchange: this._onYearRadioChange,
+					onmouseup: this._closeYearPopup
+				}),
+				v('abbr', {
+					classes: this.classes(css.yearRadioLabel)
+				}, [ `${ i }` ])
+			]));
+		}
+
+		return radios;
+	}
+
+	private _renderMonthLabel(month: number, year: number) {
+		const { monthNames, renderMonthLabel } = this.properties;
+		return renderMonthLabel ? renderMonthLabel(month, year) : `${monthNames[month].long} ${year}`;
+	}
+
+	protected render(): DNode {
+		const {
+			labelId = `${this._idBase}_label`,
+			labels,
+			month,
+			monthNames,
+			year
+		} = this.properties;
+
+		return v('div', {
+			classes: this.classes(css.datePicker)
+		}, [
+			v('div', {
+				classes: this.classes(css.topMatter),
+				role: 'menubar'
+			}, [
+				// hidden label
+				v('label', {
+					id: labelId,
+					classes: this.classes().fixed(baseCss.visuallyHidden),
+					'aria-live': 'polite',
+					'aria-atomic': 'false'
+				}, [ this._renderMonthLabel(month, year) ]),
+
+				// month trigger
+				v('button', {
+					key: 'month-button',
+					'aria-controls': `${this._idBase}_month_dialog`,
+					'aria-expanded': String(this._monthPopupOpen),
```

Comment:
Can we be consistent and use either `String(...)` or template literals or appending `''` 😄